internal/linter/check_pk_type.go

```diff
@@ -0,0 +1,69 @@
+package linter
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/skeema/skeema/internal/tengo"
+)
+
+func init() {
+	rule := Rule{
+		CheckerFunc:     TableBinaryChecker(pkTypeChecker),
+		Name:            "pk-type",
+		Description:     "Only allow primary keys to have types listed in --allow-pk-type",
+		DefaultSeverity: SeverityIgnore,
+	}
+	rule.RelatedListOption(
+		"allow-pk-type",
+		"tinyint, smallint, mediumint, int, bigint, varbinary, binary, date, datetime, timestamp, time, year, decimal",
+		"List of allowed data types for --lint-pk-type",
+		false, // permit empty list
```

Comment:
`true` (require non-empty list) seems better in this case, that way we fail loudly if the check is enabled but the corresponding allow-list is empty. This also simplifies things slightly, can remove the conditional on lines 34-36.

fwiw the only reason allow-auto-inc can be empty is that's a way of expressing "forbid all auto_increment", since some companies mandate uuid or otherwise have sharding schemes which aren't auto_increment friendly. The other linter checks with allow-lists tend to require a non-empty list.